cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandApiErrorParserUtil.java

```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.cloudbreak.cm.util;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cloudera.api.swagger.CommandsResourceApi;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiCommand;
+import com.cloudera.api.swagger.model.ApiCommandList;
+
+public class ClouderaManagerCommandApiErrorParserUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerCommandApiErrorParserUtil.class);
+
+    private ClouderaManagerCommandApiErrorParserUtil() {
+
+    }
+
+    private static Predicate<ApiCommand> commandFailed() {
+        return cmd -> {
+            if (cmd != null) {
+                if (cmd.getActive() != null && cmd.getActive()) {
+                    return true;
+                }
```

Comment:
Just to verify that my understanding is right: Is this branch for detecting commands that are still active but timed out?

Reply:
When a timeout happens on CB side, this is around 1h, the CM is still doing tasks, probably until forever, at least it did not finish for me within 6 hours. In case of timeout, we need to walk through the CM command tree and we need to check the children of the active commands, hoping that we can find one or a few that failed. E.g. in our case what we have debugged together the Hive still did something, but a few others already failed with the well-known AuditDir problem, and this gives us enough hint that probably the end-user is hitting the permission problem.

Of course, this could be improved further, e.g in case of timeout we could find the leaf Commands that are still running, and not failed yet, and we could add this information also into the error response, that would provide extra hints that what are the commands that have never reached the expected state.

There is also further improvement possibility to add this error checking mechanism inside the poller. This means while we are polling, we can detect errors and don't need to wait until timeout, and the cluster can fail faster,  but I considered this too risky since CM API is nondeterministic.

Reply:
Okay, sounds reasonable. This logic can already improve error detection significantly (and turning on cloud storage validation later will bring even more benefits).